src/install-update.ts

```diff
@@ -0,0 +1,167 @@
+
+export interface Update {
+  name: string;
+  current: string;
+  latest: string;
+  error?: {
+    message: string;
+  } | null;
+}
+
+/**
+ * updateDependencies checks for SDK-related dependency updates. If
+ * updatable releases are found, dependency files are updated with the
+ * latest dependency versions and the project changes are cached.
+ */
+export const updateDependencies = async () => {
+  const { releases } = await checkForSDKUpdates();
+  const updatableReleases = releases.filter((r) => r.current && r.latest);
+  const updateResp = await createUpdateResp(updatableReleases);
+
+  // If no errors occurred during installation, re-build
+  // project as a means to cache the changes
+  if (!updateResp.error) runBuildHook();
+
+  return updateResp;
+};
+
+/**
+ * createUpdateResp creates an object that contains each update,
+ * featuring information about the current and latest version, as well
+ * as if any errors occurred while attempting to update each.
+ */
+export async function createUpdateResp(
+  releases: Release[],
+): Promise<InstallUpdateResponse> {
+  const updateResp: InstallUpdateResponse = { name: SDK_NAME, updates: [] };
+
+  if (!releases.length) return updateResp;
+
+  try {
+    const cwd = Deno.cwd();
+
+    // Update import_map.json with latest dependency versions
+    const importsUpdateResp = await updateDependencyFile(
+      `${cwd}/import_map.json`,
+      releases,
+    );
+
+    // Update slack.json with latest dependency versions
+    const hooksUpdateResp = await updateDependencyFile(
+      `${cwd}/slack.json`,
+      releases,
+    );
+
+    updateResp.updates = [...importsUpdateResp, ...hooksUpdateResp];
+  } catch (err) {
+    updateResp.error = { message: err.message };
+  }
+
+  return updateResp;
+}
+
+/**
+ * updateDependencyFile reads the contents of a provided path, calls updateDependencyMap
+ * to swap out the current versions with the latest releases available, and then writes
+ * to the dependency file. Returns a summary of updates made.
+ */
+export async function updateDependencyFile(
+  path: string,
+  releases: Release[],
+): Promise<Update[]> {
+  try {
+    const dependencyMap = await getJSON(path);
+    const { imports, hooks } = dependencyMap;
+
+    // If file doesn't exist or expected dependency key is missing
+    if (!imports && !hooks) return [];
+
+    const dependencyKey = imports ? "imports" : "hooks";
+
+    // Update only the dependency-related key in given file ("imports" or "hooks")
+    const { updatedDependencies, updateSummary } = updateDependencyMap(
+      dependencyMap[dependencyKey],
+      releases,
+    );
+
+    // Replace the dependency-related section with the updated version
+    dependencyMap[dependencyKey] = updatedDependencies;
+    await Deno.writeTextFile(
+      path,
+      JSON.stringify(dependencyMap, null, 2).concat("\n"),
+    );
+
+    return updateSummary;
+  } catch (_) {
+    throw new Error(`An error occurred while reading a dependency file`);
```

Comment:
This is a generic error that is thrown and attached to the `Update` as the `error: message` property. Open to suggestions on how to shore this up if we don't think it's doing enough for the user.

Reply:
I would also include the error message from this catch block in the error being re-thrown, otherwise in the case that an error happens, we don't give users info as to why. And also for our future issue triaging selves, we probably want this info 😄